components/suggestion/channel_mention_provider.jsx

```diff
@@ -157,6 +158,9 @@ export default class ChannelMentionProvider extends Provider {
                 const wrappedMoreChannels = [];
                 const moreChannels = [];
                 channels.forEach((item) => {
+                    if (item.delete_at > 0 && !myMembers[item.id]) {
```

Comment:
Why do we check for `!myMembers`? Would a just archived channel still be in myMembers object?

Reply:
Yes, my memberships includes the archived channels members. The point here is to not show archived channels where you are not a member.